common/thrift_router.h

```diff
@@ -399,7 +410,8 @@ class ThriftRouter {
         // prefer local
         v.reserve(host_info.size());
         for (const auto& hi : host_info) {
-          if (role == Role::ANY || hi.second == role) {
+          if ((role == Role::ANY || hi.second == role) &&
+              (specific_az == "" || hi.first->az.compare(specific_az) == 0)) {
```

Comment:
avoid boiler plate code, consider extracting a common function for the az check, e.g.

```
bool matchAZ(const string& az, const string& specific_az) {
   return specific_az != "" && az.compare(specific_za) == 0
}
```

Reply:
done